### PR TITLE
8320278: ARM32 build is broken after JDK-8301997

### PR DESCRIPTION
**Pull request by matias9927**
JDK-8301997 changed the handling of constant pool cache entries for methods and fully removed the ConstantPoolCacheEntry class. This commit included changes to the interpreters for all supported platforms except ARM32, and its omission resulted in a GHA failure. This patch intends to introduce an ARM32 port that reflects the code changes to the included platforms so the ARM32 code can build and thus pass testing. Verified with tier 1-5 tests on Oracle platforms but not ARM32.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8320278](https://bugs.openjdk.org/browse/JDK-8320278): ARM32 build is broken after JDK-8301997 (**Bug** - P2)

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [d9db3337](https://git.openjdk.org/jdk/pull/16749/files/d9db33376affd4d6157f80806c6258b6aa2479a9)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to [d9db3337](https://git.openjdk.org/jdk/pull/16749/files/d9db33376affd4d6157f80806c6258b6aa2479a9)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16749/head:pull/16749` \
`$ git checkout pull/16749`

Update a local copy of the PR: \
`$ git checkout pull/16749` \
`$ git pull https://git.openjdk.org/jdk.git pull/16749/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16749`

View PR using the GUI difftool: \
`$ git pr show -t 16749`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16749.diff">https://git.openjdk.org/jdk/pull/16749.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16749#issuecomment-1819858536)

### CODE REVIEW

File: src/hotspot/cpu/arm/templateTable_arm.cpp [hunk 1]
```diff
@@ -2569,16 +2568,24 @@ void TemplateTable::volatile_barrier(MacroAssembler::Membar_mask_bits order_cons
...
   Label resolved;
   Bytecodes::Code code = bytecode();
-  assert(byte_no == f1_byte || byte_no == f2_byte, "byte_no out of range");
-  __ get_cache_and_index_and_bytecode_at_bcp(Rcache, Rindex, Rtemp, byte_no, 1, index_size);
+  __ load_method_entry(Rcache, Rindex);
+  switch(byte_no) {
+    case f1_byte:
+      __ add(Rtemp, Rcache, in_bytes(ResolvedMethodEntry::bytecode1_offset()));
+    case f2_byte:
+      __ add(Rtemp, Rcache, in_bytes(ResolvedMethodEntry::bytecode2_offset()));
```
Comment (coleenp) [inline src/hotspot/cpu/arm/templateTable_arm.cpp:2584]: this is missing breaks.

**Review by coleenp — APPROVED** [review]
This looks good.  This should help the arm32 porters get started and fix the compilation errors.  I think in the description, you meant to say you ran tier1-4 on Oracle platforms not arm32.

**Review by tstuefe — APPROVED** [review]
Great, thanks for fixing.

**Comment by matias9927** [comment]
Thanks for the reviews @coleenp and @tstuefe!
/integrate

**Pull request closed without merging**